plugins/preserve-rooms.cpp

```diff
@@ -446,7 +446,7 @@ static void handle_missing_assignments(color_ostream &out,
         if (!hf || hf->died_year > -1 || was_expelled(hf))
             continue;
         if (auto unit = df::unit::find(hf->unit_id)) {
-            if (Units::isActive(unit) && !Units::isDead(unit) && active_unit_ids.contains(unit->id)) {
+            if (!Units::isDead(unit) && Units::isActive(unit) && active_unit_ids.contains(unit->id)) {
```

Comment:
Moved `isActive` after `isDead`, which is a simple flag check. Not sure how `isActive` compares to testing an unordered set. The latter is probably much faster for units that are actually raiding? In that case we should test `isActive` afterwards, or maybe switch to directly testing the `inactive` flag.